sqlx-postgres/src/types/cube.rs

```diff
@@ -0,0 +1,363 @@
+}
+
+impl Type<Postgres> for PgCube {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("cube")
+    }
+}
+
+impl<'r> Decode<'r, Postgres> for PgCube {
+    fn decode(value: PgValueRef<'r>) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
+        match value.format() {
+            PgValueFormat::Text => Ok(PgCube::deserialize_string(&value.as_str()?)?),
+            PgValueFormat::Binary => {
+                let bytes = value.as_bytes()?;
+                Ok(PgCube::deserialize_bytes(bytes)?)
+            }
+        }
+    }
+}
+
+impl<'q> Encode<'q, Postgres> for PgCube {
+    fn produces(&self) -> Option<PgTypeInfo> {
+        Some(PgTypeInfo::with_name("cube"))
+    }
+
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> IsNull {
+        buf.extend_from_slice(&self.serialize());
+
+        IsNull::No
+    }
+}
+
+const BYTE_WIDTH: usize = 8;
+
+impl PgCube {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buff: Vec<u8> = vec![];
+        match self {
+            PgCube::Point(value) => {
+                buff.extend_from_slice(&[128, 0, 0, 1]);
+                buff.extend_from_slice(&value.to_be_bytes());
+            }
+            PgCube::ZeroVolume(values) => {
+                let dimension = values.len() as u8;
+                buff.extend_from_slice(&[128, 0, 0]);
+                buff.extend_from_slice(&dimension.to_be_bytes());
+                let bytes = values
+                    .into_iter()
+                    .flat_map(|v| v.to_be_bytes())
+                    .collect::<Vec<u8>>();
+                buff.extend_from_slice(&bytes);
+            }
+            PgCube::OneDimensionInterval(x, y) => {
+                buff.extend_from_slice(&[0, 0, 0, 1]);
+                buff.extend_from_slice(&x.to_be_bytes());
+                buff.extend_from_slice(&y.to_be_bytes());
+            }
+            PgCube::MultiDimension(multi_values) => {
+                let dimension = multi_values
+                    .first()
+                    .map(|arr| arr.len() as u8)
+                    .unwrap_or(1 as u8);
+                buff.extend_from_slice(&[0, 0, 0]);
+                buff.extend_from_slice(&dimension.to_be_bytes());
+                let bytes = multi_values
+                    .into_iter()
+                    .flat_map(|inner| inner)
+                    .flat_map(|v| v.to_be_bytes())
+                    .collect::<Vec<u8>>();
+                buff.extend_from_slice(&bytes);
+            }
+        };
+        buff
+    }
+
+    fn get_f64_from_bytes(bytes: &[u8], start: usize) -> Result<f64, Error> {
+        bytes[start..start + BYTE_WIDTH]
+            .try_into()
+            .map(f64::from_be_bytes)
+            .map_err(|err| Error::Decode(format!("Invalid bytes slice: {:?}", err).into()))
+    }
+    // Helper to deserialize a vector of f64 values
+    fn deserialize_vector(bytes: &[u8], start_index: usize) -> Result<Vec<f64>, Error> {
```

Comment:
As a general Rust convention, anything that doesn't take or return `self` should be a free function (at the module level), not a method.